src/core/ad.jl

```diff
@@ -26,31 +26,25 @@ end
 
 abstract type ADBackend end
 struct ForwardDiffAD{chunk} <: ADBackend end
-getchunksize(::T) where {T <: ForwardDiffAD} = getchunksize(T)
-getchunksize(::Type{ForwardDiffAD{chunk}}) where chunk = chunk
-getchunksize(::T) where {T <: Sampler} = getchunksize(T)
-getchunksize(::Type{<:Sampler{T}}) where {T} = getchunksize(T)
-getchunksize(::SampleFromPrior) = getchunksize(Nothing)
-getchunksize(::Type{Nothing}) = CHUNKSIZE[]
+getchunksize(::Type{<:ForwardDiffAD{chunk}}) where chunk = chunk
+getchunksize(::Type{<:Sampler{Talg}}) where Talg = getchunksize(Talg)
+getchunksize(::Type{SampleFromPrior}) = CHUNKSIZE[]
 
 struct TrackerAD <: ADBackend end
 
 ADBackend() = ADBackend(ADBACKEND[])
 ADBackend(T::Symbol) = ADBackend(Val(T))
 
-ADBackend(::Val{:forward_diff}) where {T} = ForwardDiffAD{CHUNKSIZE[]}
-ADBackend(::Val{T}) where {T} = TrackerAD
+ADBackend(::Val{:forward_diff}) = ForwardDiffAD{CHUNKSIZE[]}
+ADBackend(::Val) = TrackerAD
 
 """
 getADtype(alg)
 
 Finds the autodifferentiation type of the algorithm `alg`.
 """
-getADtype(::Nothing) = getADtype(Nothing)
-getADtype(::Type{Nothing}) = getADtype()
-getADtype() = ADBackend()
-getADtype(s::Sampler) = getADtype(typeof(s))
-getADtype(s::Type{<:Sampler{TAlg}}) where {TAlg} = getADtype(TAlg)
+getADtype(::Nothing) = ADBackend()
```

Comment:
Do we need this now that the algorithm cannot be `nothing`?

Reply:
I guess you're right and it can be removed :+1: